components/user-mgt/org.wso2.carbon.identity.user.pre.update.profile.action/src/main/java/org/wso2/carbon/identity/user/pre/update/profile/action/internal/execution/PreUpdateProfileResponseProcessor.java

```diff
@@ -44,6 +88,403 @@ public ActionExecutionStatus<Success> processSuccessResponse(FlowContext flowCon
+    }
+
+    private String[] filterDuplicatedValues(String claimValue, String separator) {
+
+        return Arrays.stream(claimValue.split(separator))
+                .map(String::trim)
+                .filter(s -> !s.isEmpty())
+                .toArray(String[]::new);
+    }
+
+    private void validateReplaceAndRemovePaths(List<UserClaim> claims, String path)
+            throws ActionExecutionResponseProcessorException {
+
+        Pattern pattern = Pattern.compile("^/user/claims/(\\d+)(/value/(\\d+))?");
+        Matcher matcher = pattern.matcher(path);
+
+        if (matcher.find()) {
+            int index = Integer.parseInt(matcher.group(1));
+            if (index < 0 || index >= claims.size()) {
+                throw new ActionExecutionResponseProcessorException("Invalid index for claims: " + index);
+            }
+            UserClaim claim = claims.get(index);
+            if (matcher.group(2) != null) {
+                int valueIndex = Integer.parseInt(matcher.group(3));
+                String[] values = (String[]) claim.getValue();
+                if (valueIndex < 0 || valueIndex >= values.length) {
+                    throw new ActionExecutionResponseProcessorException("Invalid index for multivalued claim: "
+                            + valueIndex);
+                }
+            }
+        } else {
+            throw new ActionExecutionResponseProcessorException("Invalid path format: " + path);
+        }
+    }
+
+    private void validateUserValuesForAddAndModifyOperations(String claimValue, String claimUri, String userId,
+                                                             Operation operation)
+            throws ActionExecutionResponseProcessorException {
+
+        UniqueIDUserStoreManager userStoreManager = getUserStoreManager();
+        try {
+            Map<String, String> userClaimValues = userStoreManager.getUserClaimValuesWithID(userId,
+                    new String[] {claimUri}, UserCoreConstants.DEFAULT_PROFILE);
+            if (operation == Operation.ADD && userClaimValues.get(claimUri) != null && !claimValue.trim().isEmpty()) {
+                throw new ActionExecutionResponseProcessorException(claimUri + " claim value already exists for user.");
+            } else if (operation.equals(Operation.REPLACE) && userClaimValues.get(claimUri) == null) {
+                throw new ActionExecutionResponseProcessorException(claimUri + " claim value does not exist for user.");
+            }
+        } catch (org.wso2.carbon.user.core.UserStoreException e) {
+            throw new ActionExecutionResponseProcessorException("Failed to retrieve user claims from user store.", e);
+        }
+    }
+
+    private void validateImmutableClaims(String claimUri) throws ActionExecutionResponseProcessorException {
+
+        if (claimUri.equals(ID_CLAIM_URI) || claimUri.equals(CREATED_CLAIM_URI) ||
+                claimUri.equals(FAMILY_NAME_CLAIM_URI) || claimUri.contains(IDENTITY_CLAIM_URI_PREFIX)) {
```

Comment:
why family name is not allowed for modifications ? Does SCIM blocks it ?
Also we can't avoid modifying all identity claims. There are certain flow related claims and claims like created, updated that should be avoided.

@ashanthamara synch up on the claim categorization along with User management team and find a way to represent claims modifiable and non modifiable.